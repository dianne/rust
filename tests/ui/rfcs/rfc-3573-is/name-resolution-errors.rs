//! Test that the lexical scope for `is`'s bindings doesn't extend farther than it should.
//@ edition: 2024
//@ aux-crate: is_macro=is-macro.rs
#![feature(builtin_syntax)]
use is_macro::is;

fn main() {
    // `is` used where `let` expressions can be used behaves like a `let` expression.
    if x //~ ERROR cannot find value `x` in this scope
        && is!(true is x)
        && x
    {
        x;
    }
    x; //~ ERROR cannot find value `x` in this scope

    // Elsewhere, `is`'s bindings only extend to the end of its `&&`-chain.
    if x //~ ERROR cannot find value `x` in this scope
        && (is!(true is x) && x)
        && x //~ ERROR cannot find value `x` in this scope
        && (is!(true is x) || x) //~ ERROR cannot find value `x` in this scope
    {
        x; //~ ERROR cannot find value `x` in this scope
    }
    is!(true is x);
    x; //~ ERROR cannot find value `x` in this scope
}
