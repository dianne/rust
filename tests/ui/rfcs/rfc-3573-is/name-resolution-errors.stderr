error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:9:8
   |
LL |     if x
   |        ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:15:5
   |
LL |     x;
   |     ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:18:8
   |
LL |     if x
   |        ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:20:12
   |
LL |         && x
   |            ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:21:31
   |
LL |         && (is!(true is x) || x)
   |                               ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:23:9
   |
LL |         x;
   |         ^ not found in this scope

error[E0425]: cannot find value `x` in this scope
  --> $DIR/name-resolution-errors.rs:26:5
   |
LL |     x;
   |     ^ not found in this scope

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0425`.
